src/pages/achievement/control/subcomponents/controlPanelTools/controlPanelUtils/AchievementSelector.tsx

```diff
@@ -0,0 +1,107 @@
+import { Button, Classes, Dialog, MenuItem } from '@blueprintjs/core';
+import { ItemRenderer, Select } from '@blueprintjs/select';
+import React from 'react';
+
+import AchievementInferencer from '../../../../dashboard/subcomponents/utils/AchievementInferencer';
+
+type AchievementSelectorProps = {
+  inferencer: AchievementInferencer;
+  dialogHeader: string;
+  selections: number[];
+  selectedId: number;
+  setSelectedId: any;
+  buttonText: string;
+  emptySelectionsMessage: string;
+  action: any;
+  isDialogOpen: boolean;
+  toggleDialogOpen: any;
+};
+
+function AchievementSelector(props: AchievementSelectorProps) {
```

Comment:
This component is not a React selector, so may be confusing to name it as such

Reply:
Renamed!